pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -296,6 +297,13 @@ func (s *Reconciler) checkControlPlaneMachines() (string, error) {
 }
 
 func coreV1Client(kubeconfig string) (corev1.CoreV1Interface, error) {
+	if kubeconfig == "" {
```

Comment:
can we unify with `coreClient`?

Reply:
unfortunately , one uses type client.Client and other corev1.CoreV1Interface, there is no easy way to convert between the 2, though both use the same RestClient.

Reply:
`coreV1Client` is used by `getNodeReference` to get  list of nodes. Is there any reason why we couldn't use `scope.CoreClient` instead and drop `coreV1Client`?
This could how ever be a different PR